076-support-for-kafka-4.0.md

```diff
@@ -0,0 +1,203 @@
+
+Once a new minor Strimzi version is released, we will stop supporting the previous Strimzi release.
+There are some exceptions, especially in the following cases:
+* Serious bugs or CVEs
+* The latest minor Strimzi release is new and some people might not have had chance to upgrade to it
+
+#### KRaft support
+
+Strimzi currently supports both ZooKeeper- and KRaft-based Apache Kafka clusters.
+It also supports migration of ZooKeeper-based clusters to KRaft.
+The current limitations to the KRaft support are based on limitations in Apache Kafka itself.
+At the time of writing this proposal, limitations include:
+* Scaling of KRaft controller nodes up or down is not supported (should be addressed in [KIP-853](https://cwiki.apache.org/confluence/display/KAFKA/KIP-853%3A+KRaft+Controller+Membership+Changes))
+* Unregistering Kafka nodes removed from the Kafka cluster (should be addressed in [KAFKA-17094](https://issues.apache.org/jira/browse/KAFKA-17094))
+
+In addition to the limitations, we are waiting for some further improvements in Apache Kafka:
+* Ability to connect to KRaft controller nodes and manage them without rolling updates
+* JBOD support in KRaft moving to GA
+
+Some of these limitations might be addressed before Apache Kafka 4.0 is released.
+But it might not apply to all of them.
+Also, after they are released in Apache Kafka, additional time might be needed to adopt  and use them in Strimzi.
+
+### Apache Kafka
+
+Apache Kafka does a new minor or major release roughly every 4 months.
+But this is subject to change when required by the circumstances.
+The current Apache Kafka releases might be expected to happen in the following order:
+* Apache Kafka 3.8.0 release (currently in the release process, expected release in July or August?)
+* Apache Kafka 3.9.0 release with the remaining missing KRaft features (shorter release cycle - expected August or September?)
+* Apache Kafka 4.0.0 release 3-4 months after the Kafka 3.9 release without ZooKeeper support (late 2024, early 2025?)
+
+The release plan is of course subject to change.
+
+Apache Kafka currently supports every minor release for 1 year.
+During this time, each minor release would typically get one or two patch releases with bug or CVE fixes.
+The 1 year period is currently expected to apply for the Apache Kafka 3.9 release that is expected to be the last with ZooKeeper support.
+
+## Motivation
+
+The move from Apache Kafka 3.x to 4.x and the related move from ZooKeeper-based Kafka clusters to KRaft is a major change for our users as well as for Strimzi itself.
+Having a good plan for how we plan to deal with it is important as it will provide clarity to Strimzi users, developers, and create a plan of the things that need to be done.
+It should also help our users to properly plan when to migrate from ZooKeeper to KRaft, as this is a major change that might require planning several months ahead of time.
+
+## Proposal
```

Comment:
While the paragraph is well written and totally understandable I was wondering if a table would be useful at the end as a kind of summary and a way to have the big picture in one shot.
You could have one line for each Strimzi release 0.4x, 0.4y and 0.4z.
Then columns specifying: Kafka versions supported, metadata supported (ZooKeeper and/or KRaft).

Reply:
Done

Reply:
I love it! Thanks!